rpc-rs/src/provider_main.rs

```diff
@@ -73,9 +73,15 @@ where
     use std::str::FromStr as _;
 
     // initialize logger
-    let log_rx = crate::channel_log::init_logger()
-        .map_err(|_| RpcError::ProviderInit("log already initialized".to_string()))?;
-    crate::channel_log::init_receiver(log_rx);
+    match crate::channel_log::init_logger() {
+        Ok(log_rx) => crate::channel_log::init_receiver(log_rx),
+        // TODO(thomastaylor312): This is brittle, but the init_logger function is public and this
```

Comment:
If we are ok with this for now, I'll create an issue before merging this PR to address this